lib/media/presentation_timeline.js

```diff
@@ -247,12 +247,22 @@ shaka.media.PresentationTimeline = class {
       return;
     }
 
-    const firstStartTime = timeline[0].start + startOffset;
-    const lastEndTime = timeline[timeline.length - 1].end + startOffset;
+    // Date.now() is in milliseconds, from which we compute "now" in seconds.
+    const now = (Date.now() + this.clockOffset_) / 1000.0;
+    // Exclude segments that are in the "future".
+    const timelineForStart = timeline.filter((segment) =>
```

Comment:
Actually this is not a segment, I think the most appropriate word would be `timeRange`